fox_scraper/fox_scraper.py

```diff
@@ -116,35 +94,60 @@ def verify_encoding(self, item_name):
         """
         this function will scrap the catalog from a specific category
-        :param soup_sub_categories: beutifulSoup object of a specific category
+        :param sub_categories: beutifulSoup object of a specific category
         :param main_cat_num: main category number
         :return: list of items for this category
         """
         items = []
-        for soup_sub_category in soup_sub_categories:
-            soup_cat_sub_category = self.get_soup_object(self.get_web_content(soup_sub_category))
-            sub_cat_num = 0
-            for link in soup_cat_sub_category.find_all(class_='box padding_hf sp_p_padding_hf center border_b border_l margin_b_1'):
-                items.append(self.parse_item(link, main_cat_num, sub_cat_num))
-            sub_cat_num += 1
+        for sub_cat_num in range(len(sub_categories)):
+            cat_sub_category = self.get_parsed_web_content(sub_categories[sub_cat_num])
+            items += self.get_sub_category_catalog(cat_sub_category, main_cat_num, sub_cat_num)
         return items
 
-    def parse_item(self, soup_item, main_cat_num, sub_cat_num):
+    def get_sub_category_catalog(self, cat_sub_category, main_cat_num, sub_cat_num):
+        """
+        this function will receive web content for fox item and returns a list of items
+        :param cat_sub_category: web content for fox item
+        :param main_cat_num: main category number
+        :return:
+        """
+        items = []
+        for link in cat_sub_category.find_all(class_='box padding_hf sp_p_padding_hf center border_b border_l margin_b_1'):
+            items.append(self.parse_item(link, main_cat_num, sub_cat_num))
+
+        return items
+
+    def parse_item(self, html_item, main_cat_num, sub_cat_num):
         """
         parse the information into Item class
         :return: Item class
         """
-        item_img_id = soup_item.find_next().get('href')
-        item_name = soup_item.find_next(class_='pname margin_hf_b').get_text()
-        item_price = soup_item.find_next(class_='price display_inline numbers').find_next(class_='display_inline').get_text()
+        item_img_id = html_item.find_next().get('href')
+        item_name = html_item.find_next(class_='pname margin_hf_b').get_text()
+        item_price = html_item.find_next(class_='price display_inline numbers').find_next(class_='display_inline').get_text()
         nums = item_price.split()
         item_price = float(float(nums[0]) + (float(nums[1])) / 100)
-        item = fox_item(item_img_id, self.main_categories[main_cat_num], self.sub_categories[main_cat_num][sub_cat_num], item_name, item_price)
+        item = FoxItem(item_img_id, self.main_categories[main_cat_num], self.sub_categories[main_cat_num][sub_cat_num], item_name, item_price)
         return item
 
+
+    # TODO - change location to new "program" function and not here
+    def send_all_items(self, catalog):
+
+        rabbit_sender = FoxSender()
```

Comment:
we can use the "with" statement to create a try-catch statement within the class (basically).
but.. we need to create __enter__() which will replace the __init__ to create the connection and __exit__() to close the connection